slang_frontend.cc

```diff
@@ -997,14 +1003,39 @@ struct ProceduralVisitor : public ast::ASTVisitor<ProceduralVisitor, true, false
 			if (!cv.as_const().as_bool())
 				break;
 
+			RTLIL::SigSpec disable;
+
+			if (disableLoop_outs.empty()) {
+				disable = RTLIL::S0;
+			} else {
+				disable = disableLoop_outs.back();
+			}
+
+			// add next disable wire
+			RTLIL::Wire *breakNextLoop = netlist.canvas->addWire(NEW_ID_SUFFIX("breakNextLoop"), 1);
+			RTLIL::Wire *breakNextLoopOred = netlist.canvas->addWire(NEW_ID_SUFFIX("breakNextLoopOred"), 1);
+			do_simple_assign(stmt.sourceRange.start(), breakNextLoop, RTLIL::S0, false);
+			netlist.canvas->addOr(NEW_ID_SUFFIX("breakNextLoopOr"), breakNextLoop, disable, breakNextLoopOred, false);
+			disableLoop_ins.push_back(breakNextLoop);
+			disableLoop_outs.push_back(breakNextLoopOred);
+
+			SwitchHelper b(current_case, vstate, disable);
```

Comment:
I think we may be able to make this into a parallel case purely by: 
 * moving the SwitchHelper out of the loop
 * passing in `RTLIL::S0` instead of the disable signal into the helper's constructor
 * doing `b.sw->parallel_case = true;`
 * using `b.branch(disable, [&]() {` later on

Reply:
I was a bit afraid of using non-static signals in the case part, but this confirms to me that it's completely fine haha!

Reply:
Yes, I was also surprised to learn it's fine a while ago

Reply:
I may be mistaken in suggesting a parallel case for this. I need to look over how parallel cases are defined in RTLIL

Reply:
Yes, `parallel_case` is wrong for this, while it allows for more than one case to be active in a switch (in the sense of RTLIL CaseRule/SwitchRule), it doesn't establish priority between the assignments from different cases (which we obviously want here for assignments from later iterations to override those from preceding). We need to do one switch per iteration like you are already doing.